preprocessing/tree_algorithm_helper.py

```diff
@@ -0,0 +1,25 @@
+from graph import *
+
+
+def has_cycle(g: Graph, vertex: Vertex, predecessor: Vertex, visited, result):
+    """
+    Recursive function to detect cycles in a graph
+
+    :param g: input graph
+    :param vertex: vertex to start from
+    :param predecessor: predecessor vertex of vertex
+    :param visited: List with visited vertices
+    :param result: list containing the "Truth Of The Tree"
+    :return: Boolean: True if the degrees are the same
+    """
+    if True in result or len(g.vertices) is len(visited):
+        return result
+    visited.append(vertex)
+
+    for v in vertex.neighbours:
+        if v in visited and v is not predecessor:
+            result[0] = True
+            break
```

Comment:
Ik snap niet waarom je lijst maakt van je boolean, want we willen maximaal 1 cycle vinden dus gewoon true returnen zou genoeg moeten zijn en die dan naar boven laten bubbelen. Dan hoef je ook niet je ingewikkelde `if True in result` te doen

Reply:
Omdat als je er een boolean zet je m in de recursie daarna nog een aantal keer overschrijft. Als je 10 vertices langs gaat en de 4de geeft een True, dan schrijven vertices 5-10 m als False. Vond dit de simpelste oplossing want je kan recursieve algoritmes niet zo maar stoppen als je ergens in de verschillende recursies een true vind.

Reply:
#66